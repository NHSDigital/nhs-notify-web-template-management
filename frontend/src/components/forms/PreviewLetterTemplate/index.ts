export * from './PreviewLetterTemplate';
export * from './server-actions';
