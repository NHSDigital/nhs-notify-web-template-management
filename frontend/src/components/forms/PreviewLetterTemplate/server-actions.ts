import {
  LetterTemplate,
  TemplateFormState,
} from 'nhs-notify-web-template-management-utils';
import { redirect, RedirectType } from 'next/navigation';
import { z } from 'zod';

export const $FormSchema = z.object({
  previewLetterTemplateAction: z.enum(['letter-edit', 'letter-submit'], {
    message: 'Select an option',
  }),
});

export async function previewLetterTemplateAction(
  formState: TemplateFormState<LetterTemplate>,
  formData: FormData
) {
  const form = Object.fromEntries(formData.entries());

  const { success, error, data } = $FormSchema.safeParse(form);

  if (!success) {
    return {
      ...formState,
      validationError: error.flatten(),
    };
  }

  if (data.previewLetterTemplateAction === 'letter-edit') {
    return redirect(`/edit-letter-template/${formState.id}`, RedirectType.push);
  }

  if (data.previewLetterTemplateAction === 'letter-submit') {
    return redirect(
      `/submit-letter-template/${formState.id}`,
      RedirectType.push
    );
  }

  throw new Error('Unknown preview letter template action.');
}
