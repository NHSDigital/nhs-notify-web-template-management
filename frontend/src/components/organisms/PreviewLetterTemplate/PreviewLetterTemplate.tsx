'use client';

import Link from 'next/link';
import PreviewTemplateDetailsLetter from '@molecules/PreviewTemplateDetails/PreviewTemplateDetailsLetter';
import content from '@content/content';
import type { LetterTemplate } from 'nhs-notify-web-template-management-utils';
import { isRightToLeft } from 'nhs-notify-web-template-management-utils/enum';
import { getBasePath } from '@utils/get-base-path';
import { Details, WarningCallout } from 'nhsuk-react-components';
import { NHSNotifyMain } from '@atoms/NHSNotifyMain/NHSNotifyMain';
import { NHSNotifyButton } from '@atoms/NHSNotifyButton/NHSNotifyButton';
import { TemplateStatus } from 'nhs-notify-backend-client';
import classNames from 'classnames';
import { NhsNotifyErrorSummary } from '@molecules/NhsNotifyErrorSummary/NhsNotifyErrorSummary';
import NotifyBackLink from '@atoms/NHSNotifyBackLink/NHSNotifyBackLink';
import { useFeatureFlags } from '@providers/client-config-provider';

type ButtonDetails = { text: string; href: string };

export function PreviewLetterTemplate({
  template,
}: Readonly<{ template: LetterTemplate }>) {
  const {
    approveProofText,
    backLinkText,
    footer,
    preSubmissionText,
    requestProofText,
    rtlWarning,
    submitText,
    validationError,
    validationErrorAction,
    virusScanError,
    virusScanErrorAction,
  } = content.components.previewLetterTemplate;

  const basePath = getBasePath();
  const { routing } = useFeatureFlags();

  const buttonMap: Record<string, ButtonDetails> = {
    NOT_YET_SUBMITTED: {
      text: submitText,
      href: `${basePath}/submit-letter-template/${template.id}`,
    },
    PROOF_AVAILABLE: {
      text: routing ? approveProofText : submitText,
      href: `${basePath}/submit-letter-template/${template.id}`,
    },
    PENDING_PROOF_REQUEST: {
      text: requestProofText,
      href: `${basePath}/request-proof-of-template/${template.id}`,
    },
  } satisfies Partial<Record<TemplateStatus, ButtonDetails>>;

  const errors: string[] = [];
  if (template.templateStatus === 'VIRUS_SCAN_FAILED') {
    errors.push(virusScanError, virusScanErrorAction);
  }

  if (template.templateStatus === 'VALIDATION_FAILED') {
    errors.push(validationError, validationErrorAction);
  }

  const continueButton = buttonMap[template.templateStatus];

  const footerText = footer[template.templateStatus] ?? [];

  return (
    <>
      <Link href='/message-templates' id='back-link' passHref legacyBehavior>
        <NotifyBackLink>{backLinkText}</NotifyBackLink>
      </Link>
      <NHSNotifyMain>
        <div className='nhsuk-grid-row'>
          <div className='nhsuk-grid-column-full'>
            {errors.length > 0 && (
              <NhsNotifyErrorSummary
                errorState={{
                  formErrors: errors,
                }}
              />
            )}
            <PreviewTemplateDetailsLetter template={template} />

            {template.templateStatus === 'PROOF_AVAILABLE' ? (
              <section className='nhsuk-u-reading-width'>
                <Details>
                  <Details.Summary>
                    {preSubmissionText.ifDoesNotMatch.summary}
                  </Details.Summary>
                  <Details.Text>
                    {(routing
                      ? preSubmissionText.ifDoesNotMatch.paragraphsApproval
                      : preSubmissionText.ifDoesNotMatch.paragraphsSubmit
                    ).map((text, i) => (
                      <p key={i}>{text}</p>
                    ))}
                  </Details.Text>
                </Details>
                <Details>
                  <Details.Summary>
                    {preSubmissionText.ifNeedsEdit.summary}
                  </Details.Summary>
                  <Details.Text>
                    <p>{preSubmissionText.ifNeedsEdit.paragraph}</p>
                  </Details.Text>
                </Details>
                <p>
                  {routing
                    ? preSubmissionText.ifYouAreHappyParagraphApproval
                    : preSubmissionText.ifYouAreHappyParagraphSubmit}
                </p>
              </section>
            ) : null}

            {footerText.length > 0 ? (
              <div
                className={classNames(
                  'preview-letter-footer',
                  'nhsuk-u-margin-bottom-6'
                )}
              >
                {footerText.map((item, i) => (
                  <p key={`footer-${i}`}>{item}</p>
                ))}
              </div>
            ) : null}

            {isRightToLeft(template.language) && (
              <div className='nhsuk-grid-row'>
                <div className='nhsuk-grid-column-two-thirds'>
                  <WarningCallout
                    data-testid='rtl-language-warning'
                    aria-live='polite'
                    className='nhsuk-u-margin-top-3'
                  >
                    <WarningCallout.Label headingLevel='h2'>
                      {rtlWarning.heading}
                    </WarningCallout.Label>
                    <p>{rtlWarning.text1}</p>
                    <p>{rtlWarning.text2}</p>
                    <p>{rtlWarning.text3}</p>
                  </WarningCallout>
                </div>
              </div>
            )}

            {continueButton && (
              <NHSNotifyButton
                data-testid='preview-letter-template-cta'
                id='preview-letter-template-cta'
                href={continueButton.href}
              >
                {continueButton.text}
              </NHSNotifyButton>
            )}
            <p>
              <Link href='/message-templates'>{backLinkText}</Link>
            </p>
          </div>
        </div>
      </NHSNotifyMain>
    </>
  );
}
