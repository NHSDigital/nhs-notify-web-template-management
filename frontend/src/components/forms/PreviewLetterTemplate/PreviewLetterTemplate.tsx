'use client';

import Link from 'next/link';
import { PreviewTemplate } from '@organisms/PreviewDigitalTemplate';
import content from '@content/content';
import {
  LetterTemplate,
  PageComponentProps,
} from 'nhs-notify-web-template-management-utils';
import { useActionState } from 'react';
import { BackLink } from 'nhsuk-react-components';
import { getBasePath } from '@utils/get-base-path';
import { useSearchParams } from 'next/navigation';
import { NHSNotifyMain } from '@atoms/NHSNotifyMain/NHSNotifyMain';
import { previewLetterTemplateAction } from './server-actions';
import { PreviewTemplateDetails } from '@molecules/PreviewTemplateDetails';

export function PreviewLetterTemplate({
  initialState,
}: Readonly<PageComponentProps<LetterTemplate>>) {
  const searchParams = useSearchParams();

  const { sectionHeading, form, backLinkText } =
    content.components.previewLetterTemplate;

  const [state, action] = useActionState(
    previewLetterTemplateAction,
    initialState
  );
  const isFromEditPage = searchParams.get('from') === 'edit';

  return (
    <>
      <BackLink href={`${getBasePath()}/manage-templates`} id='back-link'>
        Back to all templates
      </BackLink>
      <NHSNotifyMain>
        <div className='nhsuk-grid-row'>
          <div className='nhsuk-grid-column-full'>
            <PreviewTemplate
              template={initialState}
              sectionHeading={isFromEditPage ? sectionHeading : undefined}
              form={{
                ...form,
                state,
                action,
                formId: 'preview-letter-template',
                radiosId: 'previewLetterTemplateAction',
              }}
              previewDetailsComponent={
                <PreviewTemplateDetails.Letter template={initialState} />
              }
            />
            <p>
              <Link href='/manage-templates'>{backLinkText}</Link>
            </p>
          </div>
        </div>
      </NHSNotifyMain>
    </>
  );
}
