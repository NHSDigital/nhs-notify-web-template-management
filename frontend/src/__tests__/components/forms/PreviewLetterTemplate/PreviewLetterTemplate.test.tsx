'use client';

import { render, screen } from '@testing-library/react';
import { PreviewLetterTemplate } from '@forms/PreviewLetterTemplate';
import {
  LetterTemplate,
  TemplateFormState,
} from 'nhs-notify-web-template-management-utils';
import { mockDeep } from 'jest-mock-extended';
import { useSearchParams } from 'next/navigation';
import {
  Language,
  LetterType,
  VirusScanStatus,
} from 'nhs-notify-backend-client';

jest.mock('@forms/PreviewLetterTemplate/server-actions');

jest.mock('react', () => {
  const originalModule = jest.requireActual('react');

  return {
    ...originalModule,
    useActionState: (
      _: (
        formState: TemplateFormState,
        formData: FormData
      ) => Promise<TemplateFormState>,
      initialState: TemplateFormState
    ) => [initialState, '/action'],
  };
});

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(() => ({
    get: jest.fn(() => undefined),
  })),
}));

describe('Preview letter form renders', () => {
  it('matches snapshot when navigating from manage templates screen', () => {
    const container = render(
      <PreviewLetterTemplate
        initialState={mockDeep<TemplateFormState<LetterTemplate>>({
          validationError: undefined,
          name: 'test-template-letter',
          id: 'template-id',
          language: 'en',
          letterType: LetterType.Q1,
          files: {
            pdfTemplate: {
              fileName: 'file.pdf',
              currentVersion: '4C728B7D-A028-4BA2-B180-A63CDD2AE1E9',
              virusScanStatus: VirusScanStatus.PENDING,
            },
          },
        })}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when navigating from edit screen', () => {
    const mockSearchParams = new Map([['from', 'edit']]);
    (useSearchParams as jest.Mock).mockImplementation(() => ({
      get: (key: string) => mockSearchParams.get(key),
    }));

    const container = render(
      <PreviewLetterTemplate
        initialState={mockDeep<TemplateFormState<LetterTemplate>>({
          validationError: undefined,
          name: 'test-template-letter',
          id: 'template-id',
          language: 'en',
          letterType: LetterType.Q1,
          files: {
            pdfTemplate: {
              fileName: 'file.pdf',
              currentVersion: '4C728B7D-A028-4BA2-B180-A63CDD2AE1E9',
              virusScanStatus: VirusScanStatus.PENDING,
            },
            testDataCsv: {
              fileName: 'test-data.csv',
              currentVersion: '622AB7FA-29BA-418A-B1B6-1E63FB299269',
              virusScanStatus: VirusScanStatus.PENDING,
            },
          },
        })}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('matches error snapshot', () => {
    const container = render(
      <PreviewLetterTemplate
        initialState={mockDeep<TemplateFormState<LetterTemplate>>({
          validationError: {
            formErrors: [],
            fieldErrors: {
              previewLetterTemplateAction: ['Select an option'],
            },
          },
          name: 'test-template-letter',
          id: 'template-id',
          language: 'de',
          letterType: LetterType.X3,
          files: {
            pdfTemplate: {
              fileName: 'file.pdf',
              currentVersion: '4C728B7D-A028-4BA2-B180-A63CDD2AE1E9',
              virusScanStatus: VirusScanStatus.PENDING,
            },
            testDataCsv: {
              fileName: 'test-data.csv',
              currentVersion: '622AB7FA-29BA-418A-B1B6-1E63FB299269',
              virusScanStatus: VirusScanStatus.PENDING,
            },
          },
        })}
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('renders component correctly', () => {
    render(
      <PreviewLetterTemplate
        initialState={mockDeep<TemplateFormState<LetterTemplate>>({
          validationError: undefined,
          name: 'test-template-email',
          id: 'template-id',
          language: 'hi',
          letterType: LetterType.X0,
          files: {
            pdfTemplate: {
              fileName: 'file.pdf',
              currentVersion: '4C728B7D-A028-4BA2-B180-A63CDD2AE1E9',
              virusScanStatus: VirusScanStatus.PENDING,
            },
            testDataCsv: {
              fileName: 'test-data.csv',
              currentVersion: '622AB7FA-29BA-418A-B1B6-1E63FB299269',
              virusScanStatus: VirusScanStatus.PENDING,
            },
          },
        })}
      />
    );

    expect(screen.getByTestId('letter-edit-radio')).toHaveAttribute(
      'value',
      'letter-edit'
    );

    expect(screen.getByTestId('letter-submit-radio')).toHaveAttribute(
      'value',
      'letter-submit'
    );
  });
});
