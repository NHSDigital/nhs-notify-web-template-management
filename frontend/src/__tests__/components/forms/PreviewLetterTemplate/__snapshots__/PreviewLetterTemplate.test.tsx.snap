// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview letter form renders matches error snapshot 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
      id="back-link"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <div
          class="notify-confirmation-panel nhsuk-heading-l"
        >
          Template saved
        </div>
        <div
          aria-labelledby="error-summary-title"
          class="nhsuk-error-summary"
          role="alert"
          tabindex="-1"
        >
          <h2
            class="nhsuk-error-summary__title"
            data-testid="error-summary"
            id="error-summary-title"
          >
            There is a problem
          </h2>
          <ul
            class="nhsuk-list nhsuk-error-summary__list"
          >
            <li>
              <a
                href="#previewLetterTemplateAction"
              >
                Select an option
              </a>
            </li>
          </ul>
        </div>
        <h1
          class="review__heading"
          data-testid="preview-message__heading"
        >
          test-template-letter
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                Letter - German
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag"
                />
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template file
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  >
                    file.pdf
                  </p>
                </div>
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Test personalisation file
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  >
                    test-data.csv
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <form
          action="/action"
        >
          <input
            name="form-id"
            readonly=""
            type="hidden"
            value="preview-letter-template"
          />
          <input
            name="csrf_token"
            readonly=""
            type="hidden"
            value="no_token"
          />
          <div
            class="nhsuk-form-group nhsuk-form-group--error"
          >
            <fieldset
              class="nhsuk-fieldset"
            >
              <legend
                class="nhsuk-fieldset__legend nhsuk-fieldset__legend--m"
                data-testid="previewLetterTemplateAction-form__legend"
              >
                What would you like to do next?
              </legend>
              <div
                class="nhsuk-form-group nhsuk-form-group--error"
              >
                <span
                  class="nhsuk-error-message"
                  id="previewLetterTemplateAction--error-message"
                >
                  <span
                    class="nhsuk-u-visually-hidden"
                  >
                    Error: 
                  </span>
                  Select an option
                </span>
                <div
                  aria-describedby="previewLetterTemplateAction--error-message"
                  class="nhsuk-radios"
                  id="previewLetterTemplateAction"
                >
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="letter-edit-radio"
                      id="previewLetterTemplateAction-letter-edit"
                      name="previewLetterTemplateAction"
                      type="radio"
                      value="letter-edit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="previewLetterTemplateAction-letter-edit"
                      id="previewLetterTemplateAction-letter-edit--label"
                    >
                      Edit template
                    </label>
                  </div>
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="letter-submit-radio"
                      id="previewLetterTemplateAction-letter-submit"
                      name="previewLetterTemplateAction"
                      type="radio"
                      value="letter-submit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="previewLetterTemplateAction-letter-submit"
                      id="previewLetterTemplateAction-letter-submit--label"
                    >
                      Submit template
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <button
            aria-disabled="false"
            class="nhsuk-button"
            data-testid="submit-button"
            id="preview-letter-template-submit-button"
            type="submit"
          >
            Continue
          </button>
        </form>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Preview letter form renders matches snapshot when navigating from edit screen 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
      id="back-link"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <div
          class="notify-confirmation-panel nhsuk-heading-l"
        >
          Template saved
        </div>
        <h1
          class="review__heading"
          data-testid="preview-message__heading"
        >
          test-template-letter
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                Braille letter
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag"
                />
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template file
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  >
                    file.pdf
                  </p>
                </div>
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Test personalisation file
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  >
                    test-data.csv
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <form
          action="/action"
        >
          <input
            name="form-id"
            readonly=""
            type="hidden"
            value="preview-letter-template"
          />
          <input
            name="csrf_token"
            readonly=""
            type="hidden"
            value="no_token"
          />
          <div
            class="nhsuk-form-group"
          >
            <fieldset
              class="nhsuk-fieldset"
            >
              <legend
                class="nhsuk-fieldset__legend nhsuk-fieldset__legend--m"
                data-testid="previewLetterTemplateAction-form__legend"
              >
                What would you like to do next?
              </legend>
              <div
                class="nhsuk-form-group"
              >
                <div
                  class="nhsuk-radios"
                  id="previewLetterTemplateAction"
                >
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="letter-edit-radio"
                      id="previewLetterTemplateAction-letter-edit"
                      name="previewLetterTemplateAction"
                      type="radio"
                      value="letter-edit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="previewLetterTemplateAction-letter-edit"
                      id="previewLetterTemplateAction-letter-edit--label"
                    >
                      Edit template
                    </label>
                  </div>
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="letter-submit-radio"
                      id="previewLetterTemplateAction-letter-submit"
                      name="previewLetterTemplateAction"
                      type="radio"
                      value="letter-submit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="previewLetterTemplateAction-letter-submit"
                      id="previewLetterTemplateAction-letter-submit--label"
                    >
                      Submit template
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <button
            aria-disabled="false"
            class="nhsuk-button"
            data-testid="submit-button"
            id="preview-letter-template-submit-button"
            type="submit"
          >
            Continue
          </button>
        </form>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Preview letter form renders matches snapshot when navigating from manage templates screen 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/manage-templates"
      id="back-link"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <h1
          class="review__heading"
          data-testid="preview-message__heading"
        >
          test-template-letter
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <div
            class="preview"
          >
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template ID
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                template-id
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Type
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                Braille letter
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Status
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <strong
                  class="nhsuk-tag"
                />
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Template file
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  >
                    file.pdf
                  </p>
                </div>
              </div>
            </div>
            <div
              class="nhsuk-grid-row preview__row"
            >
              <div
                class="nhsuk-grid-column-one-third preview__col"
              >
                <div
                  class="preview__col_heading"
                >
                  Test personalisation file
                </div>
              </div>
              <div
                class="nhsuk-grid-column-two-thirds col"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <form
          action="/action"
        >
          <input
            name="form-id"
            readonly=""
            type="hidden"
            value="preview-letter-template"
          />
          <input
            name="csrf_token"
            readonly=""
            type="hidden"
            value="no_token"
          />
          <div
            class="nhsuk-form-group"
          >
            <fieldset
              class="nhsuk-fieldset"
            >
              <legend
                class="nhsuk-fieldset__legend nhsuk-fieldset__legend--m"
                data-testid="previewLetterTemplateAction-form__legend"
              >
                What would you like to do next?
              </legend>
              <div
                class="nhsuk-form-group"
              >
                <div
                  class="nhsuk-radios"
                  id="previewLetterTemplateAction"
                >
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="letter-edit-radio"
                      id="previewLetterTemplateAction-letter-edit"
                      name="previewLetterTemplateAction"
                      type="radio"
                      value="letter-edit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="previewLetterTemplateAction-letter-edit"
                      id="previewLetterTemplateAction-letter-edit--label"
                    >
                      Edit template
                    </label>
                  </div>
                  <div
                    class="nhsuk-radios__item"
                  >
                    <input
                      class="nhsuk-radios__input"
                      data-testid="letter-submit-radio"
                      id="previewLetterTemplateAction-letter-submit"
                      name="previewLetterTemplateAction"
                      type="radio"
                      value="letter-submit"
                    />
                    <label
                      class="nhsuk-label nhsuk-radios__label"
                      for="previewLetterTemplateAction-letter-submit"
                      id="previewLetterTemplateAction-letter-submit--label"
                    >
                      Submit template
                    </label>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <button
            aria-disabled="false"
            class="nhsuk-button"
            data-testid="submit-button"
            id="preview-letter-template-submit-button"
            type="submit"
          >
            Continue
          </button>
        </form>
        <p>
          <a
            href="/manage-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
